### ISSUE TITLE
[turborepo] Out of memory task considered successful 

### ISSUE THREAD

**Issue by dbrxnds** [vercel/turbo#6144]
### What version of Turborepo are you using?

1.10.14

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Linux

### Describe the Bug

When a task ends with a `FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory` it is still considered successful.

### Expected Behavior

The task to fail

### To Reproduce

Fairly hard to reproduce as this is something that doesn't really happen consistently with us. But in theory any task that ends up with a heap memory error should have this behavior as far as I am aware

**Comment by NicholasLYang** [comment]
Hi @dbrxnds, thanks for the issue. Could you post a run summary? To do so, pass the `--summarize` flag and turbo should return a file with your run data.

**Comment by dbrxnds** [comment]
Hi @NicholasLYang, thanks for the reply. Unfortunately this isn't happening consistently and only in our deploy process, not locally. What do you suggest?

**Comment by NicholasLYang** [comment]
Hmm, so this would require a little bit of scripting, but you could pass the `--summarize` flag, then try printing out the file or uploading it as part of your deploy process, that way on a failure, you can see the exact status of your run.

**Comment by chris-olszewski** [comment]
@dbrxnds You might be able to encourage this happening locally by using [`--max-old-space-size`](https://nodejs.org/api/cli.html#--max-old-space-sizesize-in-megabytes) and set it very low to encourage a NodeJS OOM.

**Comment by mehulkar** [comment]
Also post version of Node and npm. There was a bug with a version of recent npm where non-zero exit codes were not propagating correctly. Turborepo only really cares about exit codes, so the actual error shouldn't be making a difference? (unless this particular error has some unhandled signal that we are also getting in the way of)

**Comment by weyert** [comment]
@dbrxnds PNPM released a new version with some lifecycle improvements. Can you try again with this version? https://github.com/pnpm/pnpm/issues/7164

**Comment by Stanzilla** [comment]
> @dbrxnds PNPM released a new version with some lifecycle improvements. Can you try again with this version? [pnpm/npm-lifecycle#41](https://github.com/pnpm/npm-lifecycle/pull/41)

these were reverted because of Windows failures

**Comment by devdumpling** [comment]
This is happening for me as well. Seeing a JS Heap overflow in an older project (Next 12), and while the overflow itself is obviously an issue, turbo is marking the task as successful. 

I can try to get a repro later in a public reproduction.

**Comment by chris-olszewski** [comment]
I believe this has been resolved with the various changes/fixes to how we spawn and manage child processes along with https://github.com/vercel/turbo/pull/8113. If you encounter this with 1.13.4+ or 2.0.0+ then please reopen.

**Comment by lailo** [comment]
I'm using version `2.0.6` and I0m having this issue again when I try to do a multi stage docker build. I was using version `1.13.4` before and it worked. Just after the update it started to fail.

```
 FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
```

**Comment by Nishchit14** [comment]
I am getting the same error. Has anyone resolved it? I am using Nextjs v15.4.3 and Turbo v2.5.5. It is crashing at type checking task on `pnpm build`

**Comment by volnei** [comment]
Seems that still a bug, can someone reopen this?
